### PR TITLE
remove formerly deprecated parameters and code

### PR DESCRIPTION
**Pull request by mmoll**
- remove rest(v1) smart proxy provider and support for foreman_api gem
- remove foreman::install::repos define
- remove apipie_task parameter
- mark Fedora 24 supported instead of Fedora 21
- remove Debian 7 (wheezy) support
- remove Ubuntu 12.04 (precise) support
  - remove brightbox repository support
  - garbage-collect now unneeded alternatives dependency

This would, per definition, make the module incompatible with all Foreman releases older than 1.12, but the only real incompatibilty is the removal of Ubuntu 12.04 support. I'd open a seperate PR for documentation updates, once this got in.

### CODE REVIEW

**Comment by domcleal** [comment]
rest_v1 should be removed from the README.md too.

It might be worth deprecating rest_v2 in this PR too, or even in 5.x so we get back down to one provider?

Otherwise :+1:.

**Comment by ekohl** [comment]
In params.pp there's also a Fedora 19 conditional which I think can be removed. It then looks like the passenger_* params have the same value and can be moved out of the case statement.

**Comment by ekohl** [comment]
And ubuntu 12 can be removed from travis as well.

**Comment by domcleal** [comment]
> And ubuntu 12 can be removed from travis as well.

I think if we first remove it from metadata.json everywhere (including here) then we can remove it from modulesync to tidy up the config. Once removed from metadata it won't be running at least.

**Comment by ekohl** [comment]
>> And ubuntu 12 can be removed from travis as well.
> 
> I think if we first remove it from metadata.json everywhere (including here) then we can remove it from modulesync to tidy up the config. Once removed from metadata it won't be running at least.

Oh that's a good point.

**Comment by mmoll** [comment]
Removed the F19 conditional and move the passenger parameters out. Regarding rest_v1, as there's still the foreman_hostgriup provider using apipie, removing the v2 provider for foreman_smartproxy probably wouldn't buy us much?

File: README.md [hunk 1]
```diff
@@ -49,34 +49,6 @@ Thus 'master' will support the upcoming major version and the current stable.
...
-
-    ruby193-rubygem-foreman-tasks
-    ruby193-rubygem-ovirt_provision_plugin
-    ruby193-rubygem-puppetdb_foreman
-
-### Foreman 1.7 compatibility notes
-
-* set `apipie_task => 'apipie:cache'` as Foreman 1.7 packages didn't have
-  precompiled API docs
-* `foreman::compute::ec2` needs `package => 'foreman-compute'` on Foreman 1.7,
```
Comment (domcleal) [inline README.md]: Should we delete these `package` parameters or just leave them be?  At a minimum you should probably also remove the class/param docs saying about the Foreman 1.7/11 package names.

**Comment by domcleal** [comment]
> Regarding rest_v1, as there's still the foreman_hostgriup provider using apipie, removing the v2 provider for foreman_smartproxy probably wouldn't buy us much?

Hm, true, leave it for now I guess and perhaps we can update everything to v3 and deprecate v2 at a later date.

File: README.md [hunk 2]
```diff
@@ -49,34 +49,6 @@ Thus 'master' will support the upcoming major version and the current stable.
 The latest release (git tag, Puppet Forge) should support current and the
 previous stable release.
 
-### Foreman 1.11 compatibility notes
```
Comment (ekohl) [inline README.md]: I think it would be helpful for users if we tell them which version they should use on 1.11.

Comment (mmoll) [inline README.md]: That's right... added a note there

File: README.md (hunk 1 above)
Comment (mmoll) [inline README.md]: cleaned up the respective classes

File: manifests/compute/openstack.pp [hunk 3]
```diff
@@ -1,20 +1,12 @@
-# = Foreman OpenStack compute resource support
```
Comment (domcleal) [inline manifests/compute/openstack.pp]: This is still useful?

File: manifests/compute/rackspace.pp [hunk 4]
```diff
@@ -1,20 +1,12 @@
-# = Foreman Rackspace compute resource support
```
Comment (domcleal) [inline manifests/compute/rackspace.pp]: ditto

File: manifests/compute/ec2.pp [hunk 5]
```diff
@@ -1,20 +1,12 @@
-# = Foreman EC2 compute resource support
```
Comment (domcleal) [inline manifests/compute/ec2.pp]: ditto

**Comment by mmoll** [comment]
Added the headline also to the other compute resouces.

**Comment by domcleal** [comment]
:+1: all looks correct.

**Comment by ekohl** [comment]
Thanks!

**Pull request merged**